nasa/run.sh

```diff
@@ -1,30 +1,41 @@
-#!/bin/bash
-# This is intended for running DPS jobs.
-
-# The environment.yaml file sets up a custom conda environment called `dem`.
-# For NASA MAAP DPS, use `source activate <custom>`, not `conda activate <custom>`
-source activate dem
+#!/usr/bin/env bash
 
-# [left  bottom  right top]
-INPUT_LEFT=$1
-INPUT_BOTTOM=$2
-INPUT_RIGHT=$3
-INPUT_TOP=$4
-COMPUTE=$5
+set -xeou pipefail
 
-# Get path to this run.sh script
-basedir=$( cd "$(dirname "$0")" ; pwd -P )
-REPO_ROOT_PATH=$(dirname ${basedir})
+# This is intended for running DPS jobs.
 
-# Per NASA MAAP DPS convention, all outputs MUST be placed
-# by the algorithm into a directory called "output".
-# Once the DPS job finishes, MAAP will copy everything from "output"
-# to a directory in my-public-bucket. Everything else on the instance
-# will be destroyed.
-mkdir -p ${PWD}/output
+# `bbox` is expected to be 4 space-separated coords: left bottom right top
+bbox=$1
+# `compute` is optional, but any non-empty value will be converted to `--compute`
+compute=${2:+--compute}
 
-# Setup the environment variables. (Req'd for sardem)
-export HOME=/home/ops
+# path to the base of this repo (one level up from this script)
+basedir=$(dirname "$(dirname "$(readlink -f "$0")")")
 
-python ${REPO_ROOT_PATH}/get_dem.py --bbox ${INPUT_LEFT} ${INPUT_BOTTOM} ${INPUT_RIGHT} ${INPUT_TOP} ${COMPUTE} --out_dir ${PWD}/output
+# Per NASA MAAP DPS convention, all outputs MUST be written under a directory
+# named 'output' relative to the current working directory.  Once the DPS job
+# finishes, MAAP will copy everything from 'output' to a directory under
+# 'my-private-bucket/dps_output'. Everything else on the instance will be lost.
+outdir="${PWD}/output"
 
+# - `sardem` requires `HOME` to be set, so if it is not set, set it to `/home/ops`.
```

Comment:
Why does sardem need `HOME` to be set  ?
I ran into an issue with writing to cache in sardem can we use the `--cache-dir` flag and set it to something writable like `${PWD}/scratch` ? https://github.com/scottstanie/sardem?tab=readme-ov-file#command-line-interface

Reply:
@sujen1412, good question about `HOME`. I don't know what issue Sam originally encountered and I did not take the time to try dropping this to see if things would still work.

I'll try dropping it, and I'll also add the `--cache-dir` flag.

Reply:
@sujen1412, how did you run `sardem`? We're specifically using "COP" as the data source, so the sardem logic doesn't make use of the cache dir in that case.

Specifically, when passing `--data-source COP` to the command line, you can see that the logic returns before the line where `--cache-dir` is used. See https://github.com/scottstanie/sardem/blob/master/sardem/dem.py#L350

Further, the most recent changes don't even make a system call to the `sardem` command line, but instead directly calls the [`download_and_stitch` function](https://github.com/scottstanie/sardem/blob/master/sardem/cop_dem.py#L16), which doesn't use a cache dir.